psi4/driver/driver_cbs.py

```diff
@@ -917,65 +1078,211 @@ def return_energy_components():
         for mplevel2 in range(2, mplevel):
             VARH[f'mp{mplevel}'][f'mp{mplevel2}'] = f'MP{mplevel2} TOTAL ENERGY'
 
-    # Integrate CFOUR methods
+    # Incorporate CFOUR methods
     VARH.update(cfour_psivar_list())
+
+    # Incorporate density functionals
+    for funcname in functionals:
+        if funcname == "hf" or funcname == "scf":
+            continue
+        VARH[funcname] = {
+                         funcname: 'DFT TOTAL ENERGY',
+                f'{funcname}-dft': 'DFT TOTAL ENERGY',
+               f'{funcname}-fctl': 'DFT FUNCTIONAL TOTAL ENERGY'}
+        if 'dispersion' in functionals[funcname].keys():
+            if functionals[funcname]['dispersion']['type'] == 'nl':
+                VARH[funcname][f'{funcname}-nl'] = 'DFT VV10 ENERGY'
+            else:
+                VARH[funcname][f'{funcname}-disp'] = 'DISPERSION CORRECTION ENERGY'
+        if 'c_mp2' in functionals[funcname].keys():
+            VARH[funcname][f'{funcname}-dh'] = 'DOUBLE-HYBRID CORRECTION ENERGY'
     return VARH
     # yapf: enable
 
 
 VARH = return_energy_components()
 
 
-def _get_default_xtpl(nbasis: int, xtpl_type: str) -> Callable:
+def _get_default_xtpl(nbasis: int, xtpl_type: Union[str, None]) -> Callable:
     """ A helper function to determine default extrapolation type.
 
     Parameters
     ----------
     nbasis
         Number of basis sets
-    xtpl_type
-        {'scf', 'corl'}
-        Extrapolation type: 'scf' for the total energy, 'corl' for just the
-        correlation component.
+    xtpl_type 
+        {'scf', 'corl', 'fctl', 'dh', None}
+        Extrapolation type: 'scf' and 'fctl' for the total energy, 'corl' and 'dh'
+        for just the correlation component, None if no extrapolation requested.
 
     Returns
     -------
     Callable
         Extrapolation function to be used.
     """
 
-    if nbasis == 1 and xtpl_type in ["scf", "corl"]:
-        return xtpl_highest_1
-    elif xtpl_type == "scf":
-        if nbasis == 2:
-            return scf_xtpl_helgaker_2
-        elif nbasis == 3:
-            return scf_xtpl_helgaker_3
+    if nbasis == 1:
+        if xtpl_type in ["scf", "corl", "fctl", "dh", None]:
+            return xtpl_highest_1
+        else:
+            raise ValidationError(f"Stage treatment must be one of ['scf','corl','fctl','dh',None], not '{xtpl_type}'")
```

Comment:
Rather than have the list appear twice, just save it to a variable, and convert it to a string for the error message. Recurring issue.